### PR TITLE
Start of multilingual core field support.

### PR DESCRIPTION
**Pull request by TkTech**
Approach #1; Override package values using an IPackageController
after_view() handler.

Currently tries to use the CKAN current user locale, then english,
then _any_ that's available if English can't be found.

package_en
package_fr
search_en
search_fr

### CODE REVIEW

**Comment by TkTech** [comment]
Also depends on https://github.com/open-data/ckanext-wet-boew/pull/94 to properly hide empty package notes.

**Comment by wardi** [comment]
Yes, this looks pretty good. I'd say follow or use the logic in ckanext-scheming's [helpers.scheming_language_text](https://github.com/open-data/ckanext-scheming/blob/master/ckanext/scheming/helpers.py#L6-L31)

File: ckanext/stcndm/plugins.py [hunk 1]
```diff
@@ -79,3 +81,34 @@ def get_helpers(self):
...
+    def before_view(self, pkg_dict):
+        """
+        Ensure that (if available) the correct language strings
+        are used for core CKAN fields.
+        """
+        default_language = u'en'
+        fields_to_fluent = (
+            u'title',
+            u'notes',
+            u'name',
```
Comment (wardi) [inline ckanext/stcndm/plugins.py]: 'name' in the dataset is the unique url slug, and will never be a dict.

'name' in the resources list, however, might need to be translated

File: ckanext/stcndm/plugins.py [hunk 2]
```diff
@@ -79,3 +81,34 @@ def get_helpers(self):
...
+            u'resources'
+        )
+
+        current_language = ckan_helpers.lang()
+
+        for field in fields_to_fluent:
+            if field in pkg_dict and isinstance(pkg_dict[field], dict):
+                field_value = pkg_dict[field]
+                if not field_value:
+                    pkg_dict[field] = None
```
Comment (wardi) [inline ckanext/stcndm/plugins.py]: Why are you replacing empty dicts with None?

**Comment by TkTech** [comment]
Depends on https://github.com/open-data/ckanext-scheming/pull/60

**Comment by TkTech** [comment]
@wardi When you say resources list, do you mean the 'resources' field in the package? I ask because that field (by the time it's passed into after_view()) is an empty dict.

**Comment by wardi** [comment]
@TkTech yes, I forgot that resources aren't a thing for the ndm project

**Pull request merged**